jps/jupedsim/commands/run_simulation.py

```diff
@@ -32,14 +36,30 @@ def execute(self, args) -> int:
             log_error(f"Error gathering required files: {e}")
             return 0
         self.simulation = Simulation()
+        self.events = deque(read_events(self.simulation_files.events_json))
+
         for iteration in range(args.iterations):
             log_info(f"Simulating iteration {iteration}")
-            log_debug("Processing events")
-            self.process_events()
+            self._process_events(iteration)
             log_debug("Updating agents")
             self.simulation.compute_next_step()
         return 0
 
-    def process_events(self) -> None:
-        # TODO
-        pass
+    def _process_events(self, current_iteration: int) -> None:
+        log_debug("Processing events")
+        while self.events and self.events[0].time > current_iteration:
```

Comment:
```suggestion
        while self.events and self.events[0].time =< current_iteration:
```
shouldn't it be `=<` to ensure that the event is in the past or at the current iteration.

Reply:
done